clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -1835,43 +1970,18 @@ std::unique_ptr<lto::LTO> createLTO(
         return true;
       }
 
-      llvm::sycl::EntryPointSet EntryPoints;
-
-      for (const Function &F : M.functions()) {
-        if (llvm::module_split::isEntryPoint(F, OnlyKernelsAsEntryPoints))
-          EntryPoints.insert(const_cast<Function *>(&F));
-      }
-      // No entry points, don't proceed
-      if (EntryPoints.empty())
-        return false;
-
       if (SaveTemps)
         PreCodeGenSaveTemps(Task, M);
 
-      // TODO: Handle spec constants.
-
-      // TODO: Handle internalization of non-entry-points, we don't do it during
-      // early split anymore.
-      // One problem is that the modules are pased in as `const Module&`, and
-      // ideally we want to delete non-entry point functions, but const-casting
-      // and modifying the module seems from here seems wrong.
-
-      auto ModuleProps = llvm::sycl::computeModuleProperties(
-          M, EntryPoints, GlobalBinProps,
-          /*SpecConstsMet=*/false, /*SpecConstsMet=*/false);
-      std::string ModulePropsStr;
-      raw_string_ostream SCOut(ModulePropsStr);
-      ModuleProps.write(SCOut);
-      std::string ModuleSyms =
-          llvm::sycl::computeModuleSymbolTable(M, EntryPoints);
-      // This part is the hackiest part of this change. However, this code is
-      // run on multiple threads, so the data structures we can use are more
-      // limited. We can't use StringRef because we would need a StringSaver to
-      // keep the values around, but StringSaver is not thread safe.
-      OffloadF.getBinary()->addTmpString(ModulePropsStr);
-      OffloadF.getBinary()->addTmpString(ModuleSyms);
-      // TODO: Use SPIR-V backend instead of SPIR-V translator once the backend
-      // is mature.
+      // Use the legacy PM because eventually we will use the
+      // PreCodeGenPassesHook field of LTOConfig which requires the legacy PM.
+      legacy::PassManager PM;
```

Comment:
Since we return `false` from `PreCodeGenPassesHook` thus telling the LTO framework to not continue, `PostOptModuleHook` is not invoked. Once we use the SPIR-V backed, this should definitively be inside `PostOptModuleHook.

Reply:
> Since we return `false` from `PreCodeGenPassesHook` thus telling the LTO framework to not continue, `PostOptModuleHook` is not invoked.

According to my understanding LTOBackend.cpp code, the `PostOptModuleHook` is invoked before the `PreCodeGenPassesHook`. 

I think we need to separate codegen to SPIR-V and running "post-link" optimization passes. SYCLLinkedModuleProcessorPass can be added even into `PostImportModuleHook` as we want to run standard llvm optimizations **after** SYCL passes handle spec constants, joint matrix, etc.

Reply:
Sorry, I typo'd it, the correct response is

Since we return false from `PreCodeGenModuleHook`  thus telling the LTO framework to not continue, PreCodeGenPassesHook is not invoked.

But sure, I'll move to the SPV backend and apply this feedback